### PR TITLE
feat: Adds 2 new MongoDB AWS regions for publishing

### PR DESCRIPTION
**Pull request by maastha**
## Proposed changes

This PR updates our publishing workflow to support the two new AWS regions ca-west-1,il-central-1

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Manual QA performed:

- [ ] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [ ] Published to AWS private registry
- [ ] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [ ] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

File: .github/workflows/cfn-publish-resource.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ on:
     inputs:
       regions:
         description: "comma seperated AWS regions.Default is all regions."
-        default: "af-south-1,ap-east-1,ap-northeast-1,ap-northeast-2,ap-northeast-3,ap-south-1,ap-southeast-1,ap-southeast-2,ap-southeast-3,ca-central-1,eu-central-1,eu-north-1,eu-south-1,eu-west-1,eu-west-2,eu-west-3,me-south-1,sa-east-1,us-east-1,us-east-2,us-west-1,us-west-2,ap-south-2,ap-southeast-4,eu-central-2,eu-south-2,me-central-1"
+        default: "af-south-1,ap-east-1,ap-northeast-1,ap-northeast-2,ap-northeast-3,ap-south-1,ap-southeast-1,ap-southeast-2,ap-southeast-3,ca-central-1,eu-central-1,eu-north-1,eu-south-1,eu-west-1,eu-west-2,eu-west-3,me-south-1,sa-east-1,us-east-1,us-east-2,us-west-1,us-west-2,ap-south-2,ap-southeast-4,eu-central-2,eu-south-2,me-central-1,ca-west-1,il-central-1"
```
Comment (AgustinBettati) [inline .github/workflows/cfn-publish-resource.yaml:8]: I was hesitant in adding these as default values given that the majority of our resources will not be present in these regions until https://jira.mongodb.org/browse/CLOUDP-233717 is completed. WDYT?

Comment (lantoli) [reply .github/workflows/cfn-publish-resource.yaml:8]: i think it's ok to so new publishes go there, and at some moment we can see which ones are missing and publish

**Review by lantoli — APPROVED** [review]

File: .github/workflows/cfn-publish-resource.yaml (hunk 1 above)
Comment (marcosuma) [reply .github/workflows/cfn-publish-resource.yaml:8]: if we make this change, I guess it means that customers will be able to create a stack in `ca-west-1` and create e.g. a `stream instance` (the next one to be released) but not able to create a cluster. Is that correct? do we want such experience?

Comment (maastha) [reply .github/workflows/cfn-publish-resource.yaml:8]: >if we make this change, I guess it means that customers will be able to create a stack in ca-west-1 and create e.g. a stream instance (the next one to be released) but not able to create a cluster. Is that correct? do we want such experience?

regarding this, we already plan to publish new stream resources to the new regions (ca-west-1) anyway. If this experience seems like a concern, then that means we should hold off releasing these new resources to these regions altogether. IMO it should be fine to publish to new regions now.

**Comment by maastha** [comment]
Holding off on merging this PR for now, as we are experiencing technical issues when publishing in the new AWS regions (looks like CFN & CodeBuild are not completely supported in `ca-west-1` yet)
cc @AgustinBettati

**Comment by Zuhairahmed** [comment]
just FYI, AWS CFN is now supported service in `ca-west-1` region, but still waiting on CodeBuild - https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services/

**Pull request closed without merging**